### PR TITLE
The hook "displayProductButtons" is deprecated

### LINKED ISSUE #29189: hookDisplayProductButtons is deprecated - ps_sharebutton in PS 8.0.0
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

In PS 8.0.0, the product page with ps_sharebutton module installed displays this warning below Add to cart button
_Deprecated: The hook "displayProductButtons" is deprecated, please use "displayProductAdditionalInfo" instead in module "ps_sharebuttons"._ 

### Expected behavior

There should be no warnings because the module should not use a deprecated hook.

The module must use hook "displayProductAdditionalInfo".

### Steps to reproduce

1. Install a fresh PS 8.0.0 with sample data and all natives module.
2. Open the product page and see the warning there, under the Add to cart button.
ps_sharbutton_hookDeprecated_DisplayProductButtons_PS800

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

ps_sharebuttons 2.1.1

### PR DESCRIPTION
**Pull request by leemyongpakvn**
use "displayProductAdditionalInfo" instead

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | The hook "displayProductButtons" is deprecated, and need to be replaced by "displayProductAdditionalInfo". This brings 2 changes: it uses the new hook and it adds a migration script, so that if user upgrades module ps_sharebuttons to version 2.1.2, the script will automatically "un-hook" module from old hook and hook the module on the right hook.
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | yes
| Fixed ticket? | Fixes PrestaShop/Prestashop#29189.
| How to test?  | Please follow above issue's Step to reproduce before and after PR. Do not forget to **Reset** module in Back-office after PR.

### CODE REVIEW

**Comment by ghost** [comment]
Hello @leemyongpakvn can you add an upgrade file with unregisterhook('displayProductButtons')

**Review by PrestaEdit — CHANGES_REQUESTED** [review]
As @okom3pom said, you need to add the upgrade file to unregist/register the hook.

**Comment by leemyongpakvn** [comment]
@okom3pom n @PrestaEdit I did not see the upgrade folder so I thought this module is an exception. The upgrade file is uploaded with a new created upgrade folder now.

File: upgrade/install-2.1.2.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+function upgrade_module_2_1_2($object)
+{
+    return $object->unregisterHook('displayProductButtons')
```
Comment (ghost) [inline upgrade/install-2.1.2.php]:
```suggestion
    return $object->unregisterHook('displayProductButtons')
```

**Comment by ghost** [comment]
When the issue is ok for QA i approve

**Review by ghost — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @leemyongpakvn ,

Thanks for the PR!

Tested and approved.
No more deprecate message is shown.
Position of the module is correct.

It is QA ✅

**Comment by matthieu-rolland** [comment]
thank you @leemyongpakvn and @florine2623 !

**Pull request merged**